javascript/sbp/acquisition.js

```diff
@@ -0,0 +1,148 @@
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+// Automatically generated from piksi/yaml/swiftnav/sbp/acquisition.yaml with generate.py.
+// Please do not hand edit!
+
+/**
+ * Satellite acquisition messages from the device.
+ */
+
+var SBP = require('./sbp');
+var Parser = require('binary-parser').Parser;
+
+var MSG_ACQ_RESULT = function (sbp, msg) {
+  
+  /*SBP class for message MSG_ACQ_RESULT (0x0014).
+  
+
+  
+  This message describes the results from an attempted GPS signal
+acquisition search for a satellite PRN over a code phase/carrier
+frequency range. It contains the parameters of the point in the
+acquisition search space with the best signal-to-noise (SNR)
+ratio.
+
+
+  Parameters
+  ----------
+  snr : float
+    SNR of best point. Currently dimensonless, but will have
+units of dB Hz in the revision of this message.
+
+  cp : float
+    Code phase of best point
+  cf : float
+    Carrier frequency of best point
+  sid : int
+    Signal identifier of the satellite signal for which acquisition
+was attempted - values 0x00 through 0x1F represent GPS PRNs 1
+through 32 respectively (PRN-1 notation); other values reserved
+for future use.
+
+  sender : int
+    Optional sender ID, defaults to SENDER_ID (see sbp/msg.py).
+
+  */
+
+  this.sbp = sbp;
+  this.fields = this.parser.parse(msg);
+
+  return this;
+};
+MSG_ACQ_RESULT.prototype = Object.create(SBP.prototype);
+MSG_ACQ_RESULT.prototype.constructor = MSG_ACQ_RESULT;
+MSG_ACQ_RESULT.prototype.toBuffer = function () {
+  var buffers = [];
+  var b;
+  b = new Buffer( 4 );
+  b.writeFloatLE(this.fields.snr);
+  buffers.push(b);
+  b = new Buffer( 4 );
+  b.writeFloatLE(this.fields.cp);
+  buffers.push(b);
+  b = new Buffer( 4 );
+  b.writeFloatLE(this.fields.cf);
+  buffers.push(b);
+  b = new Buffer( 4 );
+  b.writeUInt32LE(this.fields.sid);
+  buffers.push(b);
+  return Buffer.concat(buffers);
+};
+MSG_ACQ_RESULT.prototype.parser = new Parser()
+  .endianess('little')
+  .floatle('snr')
+  .floatle('cp')
+  .floatle('cf')
+  .uint32('sid');
+
+var MSG_ACQ_RESULT_DEP_A = function (sbp, msg) {
+  
+  /*SBP class for message MSG_ACQ_RESULT_DEP_A (0x0015).
+  
+
+  
+  Deprecated.
+
+  Parameters
+  ----------
+  snr : float
+    SNR of best point. Currently dimensonless, but will have
+units of dB Hz in the revision of this message.
+
+  cp : float
+    Code phase of best point
+  cf : float
+    Carrier frequency of best point
+  prn : int
+    PRN-1 identifier of the satellite signal for which
+acquisition was attempted
+
+  sender : int
+    Optional sender ID, defaults to SENDER_ID (see sbp/msg.py).
+
+  */
+
+  this.sbp = sbp;
+  this.fields = this.parser.parse(msg);
+
+  return this;
+};
+MSG_ACQ_RESULT_DEP_A.prototype = Object.create(SBP.prototype);
+MSG_ACQ_RESULT_DEP_A.prototype.constructor = MSG_ACQ_RESULT_DEP_A;
+MSG_ACQ_RESULT_DEP_A.prototype.toBuffer = function () {
+  var buffers = [];
```

Comment:
It appears that they don't. I'm happy to make this `toBuffer` routine less tedious-looking. I already moved some of the stuff to `sbp.js` (it used to all be inlined).